OpenRA.Mods.Common/Widgets/CheckboxWidget.cs

```diff
@@ -18,73 +18,77 @@ namespace OpenRA.Mods.Common.Widgets
 {
 	public class CheckboxWidget : ButtonWidget
 	{
-		public string CheckType = "checked";
-		public Func<string> GetCheckType;
+		public new string Background = "checkbox";
+		public string Checkmark = "tick";
+		public Func<string> GetCheckmark;
 		public Func<bool> IsChecked = () => false;
-		public int CheckOffset = 2;
-		public bool HasPressedState = ChromeMetrics.Get<bool>("CheckboxPressedState");
+
+		CachedTransform<(string, bool, bool), CachedTransform<(bool, bool, bool, bool), Sprite>> getCheckmarkImageCache;
 
 		[ObjectCreator.UseCtor]
 		public CheckboxWidget(ModData modData)
 			: base(modData)
 		{
-			GetCheckType = () => CheckType;
+			GetCheckmark = () => Checkmark;
 			TextColor = ChromeMetrics.Get<Color>("TextColor");
 			TextColorDisabled = ChromeMetrics.Get<Color>("TextDisabledColor");
 			GetColor = () => TextColor;
 			GetColorDisabled = () => TextColorDisabled;
+			CreateCheckmarkImageCache();
 		}
 
 		protected CheckboxWidget(CheckboxWidget other)
 			: base(other)
 		{
-			CheckType = other.CheckType;
-			GetCheckType = other.GetCheckType;
+			Background = other.Background;
+			Checkmark = other.Checkmark;
+			GetCheckmark = other.GetCheckmark;
 			IsChecked = other.IsChecked;
-			CheckOffset = other.CheckOffset;
-			HasPressedState = other.HasPressedState;
 			TextColor = other.TextColor;
 			TextColorDisabled = other.TextColorDisabled;
 			GetColor = other.GetColor;
 			GetColorDisabled = other.GetColorDisabled;
+			CreateCheckmarkImageCache();
+		}
+
+		void CreateCheckmarkImageCache()
+		{
+			if (getCheckmarkImageCache == null)
+			{
```

Comment:
No need for the null check :)

Reply:
yes, removed before I even saw this comment